### PR TITLE
General Upgrade and BLS-to-ETH1 fixes

### PR DESCRIPTION
**Pull request by mxmar**
This PR should solve BLS-to-ETH1 CLI issues and possibly upgrade core with upstream repo.

ClickUp: https://app.clickup.com/t/2645698/DEV-5491

### CODE REVIEW

**Comment by Hugoo** [comment]
Task linked: [DEV-5491 BLS to ETH1 Withdrawal Conversion](https://app.clickup.com/t/2645698/DEV-5491)

**Comment by mxmar** [comment]
Tested with flags (non-interactive mode):
```sh
./lukso-key-gen --language=english generate-bls-to-execution-change \
--chain=lukso \
--mnemonic="sister protect peanut hill ready work profit fit wish want small inflict flip member tail between sick setup bright duck morning sell paper worry" \
--bls_withdrawal_credentials_list="0x00bd0b5a34de5fb17df08410b5e615dda87caf4fb72d0aac91ce5e52fc6aa8de,0x00a75d83f169fa6923f3dd78386d9608fab710d8f7fcf71ba9985893675d5382" \
--validator_start_index=0 \
--validator_indices="50000, 50001" \
--execution_address="0x3434343434343434343434343434343434343434";
```

And tested without flags (interactive mode):
```sh
./lukso-key-gen generate-bls-to-execution-change                                                                                                                                                                                                               ✔  1006  14:22:38

***Using the tool on an offline and secure device is highly recommended to keep your mnemonic safe.***

Please choose your language ['1. العربية', '2. ελληνικά', '3. English', '4. Français', '5. Bahasa melayu', '6. Italiano', '7. 日本語', '8. 한국어', '9. Português do Brasil', '10. român', '11. Türkçe', '12. 简体中文']:  [English]: 
Please choose the (mainnet or testnet) network/chain name ['lukso', 'lukso-testnet', 'lukso-devnet', 'mainnet', 'goerli', 'sepolia', 'zhejiang', 'holesky']:  [lukso]: 
Please enter your mnemonic separated by spaces (" "). Note: you only need to enter the first 4 letters of each word if you'd prefer.: stay depend ignore lady access will dress idea hybrid tube original riot between plate ethics ecology green response hollow famous salute they warrior little
Please enter the index position for the keys to start generating withdrawal credentials in ERC-2334 format. [0]: 9
Please enter a list of the validator index number(s) of your validator(s) as identified on the beacon chain. Split multiple items with whitespaces or commas.: 9,10,11
Please enter a list of the old BLS withdrawal credentials of your validator(s). Split multiple items with whitespaces or commas. The withdrawal credentials are in hexadecimal encoded form.: 0x00abfc563b1f859d24c91dde59b95621579792b00633b69b262d74ad1e9da3a4,0x005868054574857e54be722f367c54fbb068c50676f6abeae990146da390f0fe,0x00f0dbf34b73ba855f786a5303764ada2d2512c1ce00d52adb65408274e04448

[Error] The given input is not in hexadecimal encoded form.

Please enter a list of the old BLS withdrawal credentials of your validator(s). Split multiple items with whitespaces or commas. The withdrawal credentials are in hexadecimal encoded form.: 0x00abfc563b1f859d24c91dde59b95621579792b00633b69b262d74ad1e9da3a4,0x005868054574857e54be722f367c54fbb068c50676f6abeae990146da390f0fe,0x00f0dbf34b73ba855f786a5303764ada2d2512c1ce00d52adb65408274e04448
Please enter the 20-byte execution address for the new withdrawal credentials. Note that you CANNOT change it once you have set it on chain.: 0x3DAeE8Cd2E3C18dAfE13332DE33972AC5cF558f3

**[Warning] you are setting an Eth1 address as your withdrawal address. Please ensure that you have control over this address.**

Repeat your execution address for confirmation.: 0x3DAeE8Cd2E3C18dAfE13332DE33972AC5cF558f3

**[Warning] you are setting an Eth1 address as your withdrawal address. Please ensure that you have control over this address.**


[Error] The given Eth1 address is not in checksum form.
Please enter the 20-byte execution address for the new withdrawal credentials. Note that you CANNOT change it once you have set it on chain.: 0x3DAeE8Cd2E3C18dAfE13332DE33972AC5cF558f3

**[Warning] you are setting an Eth1 address as your withdrawal address. Please ensure that you have control over this address.**

Repeat your execution address for confirmation.: 0x3DAeE8Cd2E3C18dAfE13332DE33972AC5cF558f3

**[Warning] you are setting an Eth1 address as your withdrawal address. Please ensure that you have control over this address.**

Creating your keys:		  [####################################]  3/3
Creating your SignedBLSToExecutionChange:	  [####################################]  3/3
Verifying your BLSToExecutionChange file:	  [####################################]  3/3

Success!
Your SignedBLSToExecutionChange JSON file can be found at: /home/macht/Downloads/linux-v2.7.0-37.zip(1)/lukso-key-gen-cli-v2.7.0-37-linux/bls_to_execution_changes


Press any key.
```

- [X] Multi inputs are working
- [X] LUKSO Mainnet is the default network now (`lukso`)
- [X] Build was able to generate valid `json` message and beacon node was able to receive it
- :red_circle: [ ] In interactive mode we still have `[Error]` messages for the first time when you pass BLS withdrawals or ETH1 address
- :red_circle: [ ] There is still no solution for maximum number of the keys to change (or validator indicates). This is just a string input. It's able to collect a single key or multiple. Input length is probably limited to terminal and OS config. Linux example: https://github.com/torvalds/linux/blob/v5.11/drivers/tty/n_tty.c#L1681

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,326 @@
+version: 2.1
```
Comment (richtera) [inline .circleci/config.yml]: I had previously deleted circleci files since we use GitHub actions

File: staking_deposit/settings.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 from typing import Dict, NamedTuple
-from .version import __version__
+from eth_utils import decode_hex
 
-DEPOSIT_CLI_VERSION = __version__
+DEPOSIT_CLI_VERSION = '2.7.0'
```
Comment (richtera) [inline staking_deposit/settings.py]: The __version__ metadata s controlled by release-please so don't change that

File: staking_deposit/version.py [hunk 3]
```diff
@@ -1 +1 @@
-__version__ = '2.5.6'
+__version__ = '2.7.0'
```
Comment (richtera) [inline staking_deposit/version.py]: Do not edit manually

**Review by richtera — CHANGES_REQUESTED** [review]

File: setup.py [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 setup(
     name="staking_deposit",
-    version='2.5.6',
+    version='2.7.0',
```
Comment (richtera) [inline setup.py]: This is missing the __version__ now. Not sure what's happening

File: .release-please-manifest.json [hunk 5]
```diff
@@ -1,3 +1,3 @@
 {
-  ".": "2.5.6"
```
Comment (richtera) [inline .release-please-manifest.json]: This should not be manually edited. It's used by release please to find the previous version

File: staking_deposit/settings.py [hunk 6]
```diff
@@ -1,5 +1,5 @@
 from typing import Dict, NamedTuple
-from .version import __version__
```
Comment (richtera) [inline staking_deposit/settings.py]: Ok I am not sure I am seeing the right commits here but this delete also needs to be undone

**Review by richtera — CHANGES_REQUESTED** [review]
Hmm there are some more places the version stuff got affected

File: staking_deposit/settings.py (hunk 6 above)
Comment (richtera) [reply staking_deposit/settings.py]: Please rever this change to that on line 4` __version__` is available.

File: setup.py [hunk 7]
```diff
@@ -6,7 +6,7 @@
 
 setup(
     name="staking_deposit",
-    version='2.5.6',
+    version=__version__,
```
Comment (richtera) [inline setup.py:10]: The import of `__version__` also seems to have gotten removed.

**Review by richtera — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (mxmar) [reply .circleci/config.yml]: Agreed.

File: staking_deposit/settings.py (hunk 2 above)
Comment (mxmar) [reply staking_deposit/settings.py]: Solved

File: staking_deposit/version.py (hunk 3 above)
Comment (mxmar) [reply staking_deposit/version.py]: OK, solved.

File: setup.py (hunk 4 above)
Comment (mxmar) [reply setup.py]: I've added `__version__` here.

File: staking_deposit/settings.py (hunk 6 above)
Comment (mxmar) [reply staking_deposit/settings.py]: Done, thanks!

File: .release-please-manifest.json (hunk 5 above)
Comment (mxmar) [reply .release-please-manifest.json]: Done

File: setup.py (hunk 4 above)
Comment (mxmar) [reply setup.py]: Restored `develop` commit there.

File: setup.py (hunk 7 above)
Comment (mxmar) [reply setup.py:10]: Solved, added: `from staking_deposit import __version__`

**Review by richtera — APPROVED** [review]
NICE!

**Comment by mxmar** [comment]
I'm closing due to @richtera upgrades and fixes in https://github.com/lukso-network/tools-key-gen-cli/pull/38 and https://github.com/lukso-network/tools-key-gen-cli/pull/40

**Pull request closed without merging**